Add documentation #17

otselnik opened the pull request.

anton-lisanin commented:
Remove `clippy::missing_errors_doc, clippy::missing_panics_doc` from lib.rs. And run clippy again.

vakond commented on evm_loader/program/src/account_data.rs:
```diff
@@ -137,8 +181,10 @@ impl AccountData {
 }
 
 impl Account {
+    /// Account stuct serrialized size
```
typo: "stuct" -> struct, "serrialized" -> serialized

vakond commented on evm_loader/program/src/account_data.rs:
```diff
@@ -137,8 +181,10 @@ impl AccountData {
+    /// Account stuct serrialized size
     pub const SIZE: usize = 20+1+8+32+1+32;
 
+    /// Deserrialize `Account` struct from input data
```
"Deserrialize" -> Deserialize

vakond commented on evm_loader/program/src/account_data.rs:
```diff
@@ -152,6 +198,7 @@ impl Account {
         }
     }
 
+    /// Serrialize `Account` struct into given destination
```
"Serrialize" -> Serialize

vakond commented on evm_loader/program/src/solana_backend.rs:
```diff
@@ -17,33 +18,50 @@ use crate::{
 };
 
+/// Account storage
+/// Trait to acces account info
```
typo "acces" -> access

vakond commented:
I've tried `cargo doc --release` in directory `evm_loader/program`, and got error:
```
Documenting evm-loader v2.0.8 (/home/walk/Downloads/neon-evm/evm_loader/program)
error: unresolved link to `WRITE`
  --> src/instruction.rs:19:15
   |
19 |     ///   0. [WRITE] Account to write to
   |               ^^^^^ no item named `WRITE` in scope
```

otselnik commented:
> 
> 
> Remove `clippy::missing_errors_doc, clippy::missing_panics_doc` from lib.rs. And run clippy again.

Removed.



> 
> 
> I've tried `cargo doc --release` in directory `evm_loader/program`, and got error:
> 
> ```
> Documenting evm-loader v2.0.8 (/home/walk/Downloads/neon-evm/evm_loader/program)
> error: unresolved link to `WRITE`
>   --> src/instruction.rs:19:15
>    |
> 19 |     ///   0. [WRITE] Account to write to
>    |               ^^^^^ no item named `WRITE` in scope
> ```

Fixed

s-medvedev commented on evm_loader/program/src/account_storage.rs:
```diff
@@ -40,6 +45,14 @@ impl<'a> ProgramAccountStorage<'a> {
+    /// `ProgramError::InvalidArgument` if account in `account_infos` is wrong or in wrong place
+    /// `ProgramError::InvalidAccountData` if account's data doesn't meet requirements
+    /// `ProgramError::NotEnoughAccountKeys` if `account_infos` doesn't meet expectations
+    #[allow(clippy::missing_panics_doc)]
```
Why do you need `missing_panics_doc` here?

otselnik replied:
Clippy's false positive panic triggered because of unwrap on line 165 - it could cause panic. But I've checked it already to be Some at line 155.

otselnik replied:
I've made changes to resolve this situation and remove  allowance of `missing_panics_doc`
